### PR TITLE
refactor: use `chrono::DateTime::from_timestamp`

### PR DESCRIPTION
**Pull request by iuioiua**
`chrono::NaiveDateTime::from_timestamp_opt()` was deprecated in https://github.com/chronotope/chrono/pull/1473.

Prerequisite for #23272.

### CODE REVIEW

File: Cargo.lock
Comment (bartlomieju) [inline Cargo.lock:1]: Woah, this look concerning... Did you regenerate the whole lock file?

File: tests/wpt/runner/expectation.json
Comment (bartlomieju) [inline tests/wpt/runner/expectation.json:1]: Why these changes?

Comment (iuioiua) [reply tests/wpt/runner/expectation.json:1]: Likely because of the updated `Cargo.lock`. Reverted now.

File: Cargo.lock
Comment (iuioiua) [reply Cargo.lock:1]: Reverted

**Review by bartlomieju — APPROVED** [review]
